packages/typespec-ts/test/modularIntegration/encodeBytes.spec.ts

```diff
@@ -167,4 +169,132 @@ describe("EncodeBytesClient Rest Client", () => {
       }
     });
   });
+
+  describe.skip("request body", () => {
+    const pngFile = readFileSync(
+      resolve("../../packages/typespec-ts/temp/assets/image.png")
+    );
+    it(`should post bytes`, async () => {
+      try {
+        const result = await client.requestBody.default(
+          stringToUint8Array("dGVzdA==", "base64"),
+          {
+            requestOptions: { headers: { "content-type": "application/json" } }
```

Comment:
we need to specify contentType as application/json if the body is string type. otherwise, cadl ranch will report parse error from raw body parser. in autorest testserver case, we will generate a contentType parameter like 
```
export interface StringPutNullMediaTypesParam {
  /** Request content type */
  contentType?: "application/json";
}
```
@MaryGao do you have any pointer on why we could generate the contentType in this case but not from the TypeSpec ? Thank you.

Reply:
We don't have this kind of filter for tsp input. There is no contentType defined in tsp so we will not generate it for them.

I think the issue is introduced in core https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/core/core-client-rest/src/sendRequest.ts#L157-L174 when getting body content.

If we specify contentType the core would format as JSON.stringify(body) in our case it would be `'"dGVzdA=="'`; it not, it would be pure string that is `"dGVzdA=="`. 

The former would be a correct json but the latter would not, so cadl-ranch throws exceptions.